crates/biome_js_analyze/src/lint/nursery/no_document_import_in_page.rs

```diff
@@ -0,0 +1,94 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic, RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_js_syntax::{JsFileSource, JsImport};
+use biome_rowan::AstNode;
+
+declare_lint_rule! {
+    /// Prevents importing `next/document` outside of `pages/_document.jsx` in Next.js projects.
+    ///
+    /// The `next/document` module is intended for customizing the document structure globally in Next.js.
+    /// Importing it outside of `pages/_document.js` can cause unexpected behavior and break certain features of the framework.
+    ///
+    /// ## Examples
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// import { Document, Html } from 'next/document'
+    ///
+    /// export default class MyDocument extends Document {
+    ///   render() {
+    ///     return (
+    ///       <Html lang="en">
+    ///         {/* */}
+    ///       </Html>
+    ///     )
+    ///   }
+    /// }
```

Comment:
nits
```suggestion
    /// ```jsx
    /// import { Document, Html } from 'next/document'
    ///
    /// export default class MyDocument extends Document {
    ///   render() {
    ///     return (
    ///       <Html lang="en">
    ///         {/* */}
    ///       </Html>
    ///     )
    ///   }
    /// }
    /// ```
```